### PR TITLE
#eslint task fixes

### PR DESCRIPTION
**Pull request by AlbionaHoti**
@sarah-clements, I did a PR solving different files from the two rules:

Fix files from ```jsx-a11y/control-has-associated-label``` eslint rule.
Fix files from ```jsx-a11y/no-static-element-interactions``` eslint rule.

Can you review them? 
Please let me know if I can improve the fixes more.
This PR enables and solves the ESlint rules of 16 different files.

### CODE REVIEW

**Comment by AlbionaHoti** [comment]
@sarah-clements I did a PR solving different files from the two rules I mentioned above. Can you review them? Please let me know if I can improve the fixes more.

File: ui/job-view/details/PinBoard.jsx [hunk 1]
```diff
@@ -403,6 +401,8 @@ class PinBoard extends React.Component {
                         ? 'btn-lg selected-job'
                         : 'btn-xs'
                     }`}
+                    role="button"
```
Comment (sarah-clements) [inline ui/job-view/details/PinBoard.jsx]: There are many elements in these files that might be best changed to a `Button` component using the react-strap library we have installed. So I suggest trying to change spans and divs to buttons while trying to keep the styling of the element as similar as possible.

File: ui/job-view/pushes/PushList.jsx [hunk 2]
```diff
@@ -170,6 +169,7 @@ class PushList extends React.Component {
           <div
             className="progress active progress-bar progress-bar-striped"
             role="progressbar"
+            aria-label="list-push"
```
Comment (sarah-clements) [inline ui/job-view/pushes/PushList.jsx]: I think @camd can help you with a clearer name for what this is doing. We want this to make sense to users of screen readers.

File: ui/perfherder/compare/CompareTable.jsx [hunk 3]
```diff
@@ -120,11 +118,11 @@ export default class CompareTable extends React.PureComponent {
             </th>
             <th className="table-width-lg">Base</th>
             {/* empty for less than/greater than data */}
-            <th className="table-width-sm" />
+            <th className="table-width-sm" aria-label="first-emptiness" />
```
Comment (sarah-clements) [inline ui/perfherder/compare/CompareTable.jsx]: These aria-labels are important for screen readers so use the comments to give them descriptive names. If you are not certain how what they are for, please inspect the elements in your browser or ask us :)

File: ui/perfherder/graphs/GraphsContainer.jsx [hunk 4]
```diff
@@ -265,7 +264,12 @@ class GraphsContainer extends React.Component {
           }`}
           ref={this.tooltip}
         >
-          <span className="close mr-3 my-2 ml-2" onClick={this.closeTooltip}>
+          <span
```
Comment (sarah-clements) [inline ui/perfherder/graphs/GraphsContainer.jsx]: Same as my comment above it is probably better to use a Button element where possible.

**Pull request closed without merging**